lib/ansible/modules/packaging/os/yum.py

```diff
@@ -123,6 +128,15 @@
     default: "/"
     aliases: []
 
+  allow_downgrade:
+    description:
+      - Specify if the named package and version is allowed to downgrade
+        a maybe already installed higher version of that package
+    required: false
+    default: "no"
+    choices: ["yes", "no"]
+    version_added: "2.3"
```

Comment:
After much discussion, we've decided that the overall design here (with a transaction for install and a transaction for downgrade) is okay.  But the documentation is very important.  So this entry needs to call out that specifying ```allow_downgrade=True``` means that the task is no longer idempotent.  The task could end up with a set of packages that does not match the complete list of specified packages to install (because dependencies between the downgraded package and others can cause changes to the packages which were in the earlier transaction). So please call that out here.  There name: field should also get some more documentation but I'll call it out there.

You can also update the version_added: to "2.4" here.